### PR TITLE
Add mount as part of the unique key for roles

### PR DESCRIPTION
**Pull request by OliviaHY**
### What:
* Add its own validate uniqueness function for roles so it can allow duplicated name for different mount
* Defer to caller in main to log fatal error and abort the process
* Allow /cmd/vault-manager to be git recognized

### Why
Roles can be of different mount but same name

### Validation:
Dry-run with roles that of different mount and duplicated names, the integration ran through;
Dry-run with roles that of the same mount and duplicated names, the integration failed;

### CODE REVIEW

File: toplevel/role/role.go [hunk 1]
```diff
@@ -303,3 +304,18 @@ func addOptionalOidcDefaults(instance string, roles []entry) {
 		}
 	}
 }
+
+func ValidateRoleUniqueness(desiredRoles []entry, toplevel string) error {
+	var uniqueNames = make(map[string]bool)
+	for _, role := range desiredRoles {
+		roleName := role.Name
+		uniqueRoleKey := strings.Join([]string{roleName, role.Mount}, "")
```
Comment (dwelch0) [inline toplevel/role/role.go:312]:
```suggestion
		uniqueRoleKey := fmt.Sprintf("%s/%s", role.Mount, role.Name)
```

Comment (dwelch0) [reply toplevel/role/role.go:312]: This way we'll match vault cli's format of `auth_mount/role_name`

Comment (OliviaHY) [reply toplevel/role/role.go:312]: I'm not sure why the `/` is necessary? The mount itself usually come with a slash

Comment (OliviaHY) [reply toplevel/role/role.go]: I updated without the slash in the middle. Tested that it's not needed to result in auth_mount/role_name

File: toplevel/role/role.go [hunk 2]
```diff
@@ -303,3 +304,18 @@ func addOptionalOidcDefaults(instance string, roles []entry) {
 		}
 	}
 }
+
+func ValidateRoleUniqueness(desiredRoles []entry, toplevel string) error {
```
Comment (dwelch0) [inline toplevel/role/role.go:308]:
```suggestion
func validateRoleUniqueness(desiredRoles []entry, toplevel string) error {
```

Comment (dwelch0) [reply toplevel/role/role.go:308]: small nit so a function unique to role isn't exported by package

Comment (OliviaHY) [reply toplevel/role/role.go:308]: TIL!

File: toplevel/role/role.go [hunk 3]
```diff
@@ -303,3 +304,18 @@ func addOptionalOidcDefaults(instance string, roles []entry) {
...
+	var uniqueNames = make(map[string]bool)
+	for _, role := range desiredRoles {
+		roleName := role.Name
+		uniqueRoleKey := fmt.Sprintf("%s%s", role.Mount, role.Name)
+		_, exist := uniqueNames[uniqueRoleKey]
+		if !exist {
+			uniqueNames[uniqueRoleKey] = true
+		} else {
+			return fmt.Errorf("name %s already exist in %s", roleName, toplevel)
```
Comment (dwelch0) [inline toplevel/role/role.go:317]:
```suggestion
			return fmt.Errorf("name %s already exist in %s", role.Name, toplevel)
```

File: toplevel/role/role.go [hunk 4]
```diff
@@ -173,15 +174,15 @@ func (c config) Apply(address string, entriesBytes []byte, dryRun bool, threadPo
 		return err
 	}
 
-	desiredItems := asItems(instancesToDesiredRoles[address])
-	validateUniquenessError := vault.ValidateUniqueness(desiredItems, toplevelName)
+	desiredRoles := instancesToDesiredRoles[address]
+	validateUniquenessError := ValidateRoleUniqueness(desiredRoles, toplevelName)
```
Comment (dwelch0) [inline toplevel/role/role.go:178]: We can run this check earlier in the function. Move to L115 (call right after `instancesToDesiredRoles` is created)

File: toplevel/role/role.go [hunk 5]
```diff
@@ -173,15 +174,15 @@ func (c config) Apply(address string, entriesBytes []byte, dryRun bool, threadPo
...
 	if validateUniquenessError != nil {
 		log.Fatalln(validateUniquenessError)
```
Comment (dwelch0) [inline toplevel/role/role.go:180]:
```suggestion
		return validateUniquenessError
```

Comment (dwelch0) [reply toplevel/role/role.go:180]: Allow `main` to handle error

**Review by dwelch0 — APPROVED** [review]
LGTM!

**Pull request merged**